numpy/core/tests/test_multiarray.py

```diff
@@ -9046,17 +9046,68 @@ class TestFormat:
 
     def test_0d(self):
         a = np.array(np.pi)
-        assert_equal('{:0.3g}'.format(a), '3.14')
-        assert_equal('{:0.3g}'.format(a[()]), '3.14')
+        # general format 'g' should behave the same as Python
+        # ref: https://docs.python.org/3/library/string.html#format-specification-mini-language
+        assert_equal('{:.4g}'.format(a), '3.142')
+        assert_equal('{:.4g}'.format(10 * a), '31.42')
+        assert_equal('{:.4g}'.format(100 * a), '314.2')
+        assert_equal('{:.4g}'.format(1_000_000 * a), '3.142e+06')
+        assert_equal('{:.4e}'.format(1_000_000 * a), '3.1416e+06')
 
     def test_1d_no_format(self):
         a = np.array([np.pi])
         assert_equal('{}'.format(a), str(a))
 
     def test_1d_format(self):
-        # until gh-5543, ensure that the behaviour matches what it used to be
-        a = np.array([np.pi])
-        assert_raises(TypeError, '{:30}'.format, a)
+        # gh-5543
+        fmtspc = "+.4f"
+
+        a = np.array([-1, 0, 1, 2], dtype="int")
+        assert_equal(format(a, fmtspc), np.array_format(a, fmtspc))
+
+        a = np.array([-np.pi, np.pi],  dtype="float")
+        assert_equal(format(a, fmtspc), np.array_format(a, fmtspc))
+
+        a = np.array([-np.pi, np.pi],  dtype="float128")
```

Comment:
Windows does not have `float128`
```suggestion
        a = np.array([-np.pi, np.pi],  dtype=np.longdouble)
```